packages/core/test/app/client/wallet-db.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { should } from '../../setup'
+
+/* External Imports */
+const memdown = require('memdown')
+
+/* Internal Imports */
+import { Keystore } from '../../../src/interfaces'
+import { BaseDB } from '../../../src/app/common/db'
+import { DefaultWalletDB } from '../../../src/app/client/wallet-db'
+
+const keystore: Keystore = {
```

Comment:
Have you taken a look at `eth-crypto` specifically [this (encryption)](https://github.com/pubkey/eth-crypto#encryptwithpublickey) and [this (create identity)](https://github.com/pubkey/eth-crypto#createidentity) for keystore creation / encryption? According to https://github.com/ethers-io/ethers.js/pull/233 it's used in ethers.js.

I ask because this config looks pretty low level & I hope there are higher level libraries that should work well enough

Reply:
Or I know there are even easier approaches if we're willing to accept the web3 bloat--https://web3py.readthedocs.io/en/stable/web3.eth.account.html#extract-private-key-from-geth-keyfile

Reply:
Yeah I wouldn't create this manually - just did it here for the sake of testing. `Wallet` (not implemented in this PR) will use a library (probably eth-crypto) to generate keystore files.